### DISCUSSION TITLE
Welcome to javafx_jdbc Discussions!

### DISCUSSION THREAD

**Discussion by makklays** [makklays/javafx_jdbc#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by upircloud** [comment]
Do not keep inner classes in main class file, and at all. See https://www.baeldung.com/java-inner-classes-vs-subclasses

**Reply by makklays** [reply]
ok) edited and commited

**Comment by upircloud** [comment]
Task:
Find solution for keeping connecting to DB, and evict approach of connecting every time when you want to apply sql query.

**Reply by makklays** [reply]
add a new: 
public class dbmsConnection 
for this solution and thinking about pattern Singleton at Java ) for its implementation..

**Reply by makklays** [reply]
I made merge branch 'feature/add-mysql-driver' in branch 'master'

**Reply by makklays** [reply]
added class 'dbmsConnection' (pattern 'Singleton') - checked have only 1 instance

**Reply by upircloud** [reply]
Great, now try to use some connection pool like c3p0.

**Comment by upircloud** [comment]
Task:
1. Sort classes into packages for beautifying app structure. like i did it for User.class
2. Try to use db connection pool like c3p0 to improve performance
3. Try to use logger like logback or log4j or slf4j for logging in application instead Sytem.out.println.

**Reply by upircloud** [reply]
Please do it in PR then it will be easy to comment and discuss particular line of code.

**Reply by makklays** [reply]
What is PR ?

**Reply by makklays** [reply]
1. Done
AND add java.util.loggin.* and logging.properties

**Reply by upircloud** [reply]
> 1. Done 
>    AND add java.util.loggin.* and logging.properties
1 -> not only mdels
2 -> logback better see PR

**Comment by upircloud** [comment]
> What is PR ?

Pull request

**Reply by makklays** [reply]
at your branch ? or create new branch for it ?

**Comment by makklays** [comment]
Added 'Hibernate' to this project (and relation 'OneToMany' and 'ManyToOne'):
https://github.com/makklays/javafx_jdbc/blob/master/src/main/java/com/example/demo3/dao/ChannelEntity.java
https://github.com/makklays/javafx_jdbc/blob/master/src/main/java/com/example/demo3/dao/CompanyEntity.java

Use on the classes (for examples):
https://github.com/makklays/javafx_jdbc/blob/master/src/main/java/com/example/demo3/LoginController.java  -  (Line 78)
https://github.com/makklays/javafx_jdbc/blob/master/src/main/java/com/example/demo3/InfoController.java     -  (Line 69)

**Reply by upircloud** [reply]
Please look at PR hibernate. I dont find implemntation for LoginController(it's commented).
Also please! :
1. Cleanup the code! remove all commmented code, remove all not needed classes or resources, rename classes/variables methods according JAVA convention https://www.theserverside.com/feature/Java-naming-conventions-explained

2.When you add some new features, please do it via new feature branch also create pull request(PR), for me it will be easy to review your code.

**Reply by makklays** [reply]
1. Hize todo de esta lista 
2. Disculpe, olvidí. Entendí como hacer PR y hare PR en siguente vez ¿Vale?=)

Y por supuesto estoy añadido Hibernate para este projecto.

**Reply by makklays** [reply]
estoy leyendo sobre Hibernate 
https://habr.com/ru/articles/271115/

**Reply by makklays** [reply]
PR 'Hibernate':   https://github.com/makklays/javafx_jdbc/pull/new/feature/hibernate-config  =)

**Comment by makklays** [comment]
Other idea )
perhaps we can implement it today? after 19:00 )))
https://dou.ua/forums/topic/38358/

**Reply by makklays** [reply]
![изображение](https://github.com/makklays/javafx_jdbc/assets/12418074/7b273988-bb27-4fdc-950f-ee59d0bbc0eb)
He añadido Hibernate pero tengo un errore en la clase HibernateSessionFactory con configure() de hibernate.cfg.xml
  =(

**Reply by makklays** [reply]
![изображение](https://github.com/makklays/javafx_jdbc/assets/12418074/74f7b741-6781-483a-a071-21b24df15193)

**Reply by makklays** [reply]
PR 'Hibernate': https://github.com/makklays/javafx_jdbc/pull/new/feature/hibernate-config  =)

**Reply by makklays** [reply]
Y leo este enlace 
'5. Create a Hibernate configuration file - Java Configuration'
https://www.javaguides.net/2018/11/hibernate-query-language-insert-update.html 
Creo que eso es bueno ejemplo sobre 'Hibernate'

**Comment by makklays** [comment]
How I can see full sql-query ? What function ?

PreparedStatement stmt = con.prepareStatement(sql);
stmt.setString(1, "%"+serachtxt+"%");
stmt.setString(2, serachtxt);
System.out.println(stmt.getSQL()); // <----- get query - sql with params ? what function ?
ResultSet rs = stmt.executeQuery();
